README.md

```diff
@@ -0,0 +1,65 @@
+## Abstract
+This ARC contains is an implementation of [ARCX](https://github.com/algorandfoundation/ARCs/pull/229). The goal is to provide users a way to delegate asset opt-ins while having a high degree of control compared to a standalone logic signature without an application. The expectation is that a single instance of this ARC will be deployed on Algorand networks. 
+
+## Motivation
+[ARCX](https://github.com/algorandfoundation/ARCs/pull/229) provides a standard for delegated asset opt-ins, but there are some UX problems that need to be addressed. First, there needs to be a way to control whether the lsig is still functionality or not. Without an application to control this, signing the logic signature program would be irreversible without rekeying. There also needs to be a standardized way to store and read signatures for a given account so dApps and other users can take advantaged of delegated opt-ins.
+
+## Specification
+This is an implementation of [ARCX](https://github.com/algorandfoundation/ARCs/pull/229). There is additional functionality provided by the application in this ARC in the methods of the application.
+
+For all methods, refer to the [ABI JSON description](./contracts/artifacts/DelegatedOptIn.abi.json) for complete descriptions of arguments.
+
+### Signature Storage
+
+`setSignature(byte[64],address,txn)void` is a method that allows a user to upload their signature to box storage
+
+
+### Opt-Ins
+
+`delegatedOptIn(pay,axfer)void` is an implementation of the [ARCX](https://github.com/algorandfoundation/ARCs/pull/229) interfaces. It verifies the MBR payment is sent to the account opting in and that it covers the ASA minimum balance requirement.
+
+### Storage
+
+| Type | Key | Description |
+| ---- | --- | ----------- |
+| Box | `auth-addr` | Mapping of signer to lsig signature |
+
+## Rationale
+Box storage is used to store signatures indefinitely and [the verifier lsig](./contracts/verifier_lsig.teal) ensures the signature is always correct.
+
+End time functionality is provided to allow users to revert the effects of signing the logic signature programs without having to rekey. It also lets users open their account for opt-in delegations for a short amount of time without having to remember to manually undo it.
+
+Asset MBR is stored in global storage in case the MBR for asset were to ever change.
+
+## Backwards Compatibility
+N/A
+
+## Test Cases
+Tests written with Jest and algokit for the contract and logic signatures can be seen in [test.ts](./tests/test.ts).
+
+```
+  Delegated Opt In App
+    create
+      ✓ creates the app (765 ms)
+    setSignature
+      ✓ works with valid signature and lsig (790 ms)
+    delegatedOptIn
+      ✓ works with valid lsig and method call (1453 ms)
+    unsetSignature
```

Comment:
How about "revokeSignature" instead?

Reply:
Yeah I like that better. Changed in 89e1515a7c69e88f5370735d8d516379549fbc77